include/pistache/async.h

```diff
@@ -171,13 +171,15 @@ namespace Async {
 
         struct Core {
             Core(State _state, TypeId _id)
-                : state(_state)
+                : allocated(false)
+                , state(_state)
                 , exc()
                 , mtx()
                 , requests()
                 , id(_id)
             { }
 
+            bool allocated;
```

Comment:
Can this variable be accessed from different threads?

Reply:
If it can be accessed from different threads then that means that: 
- 'storage' can be accessed as well from different threads

That means that object in storage may be used before it is fully constructed...
So, there may be bigger issues if that is the case.

I can definitely change it to an 'atomic_bool' if you want but there are bigger issues if that's the case.

Reply:
Ok, let it be so, we can check it using thread sanitizer further

Reply:
ok, thanks, I can help debug any further issues anyway.

PS: This whole file seems a bit over-engineered, simplicity is better when doing multi-threaded.